pyfar/dsp/dsp.py

```diff
@@ -1488,3 +1488,232 @@ def decibel(signal, domain='freq', log_prefix=None, log_reference=1):
             " or 'freq_raw'.")
     data[data == 0] = np.finfo(float).eps
     return log_prefix * np.log10(np.abs(data) / log_reference)
+
+
+def soft_limit_spectrum(signal, limit, knee, ratio=np.inf, freq_range=None,
+                        direction='above'):
+    """
+    Apply a frequency-wise soft limiting to a spectrum.
+
+    Soft limiting denotes
+    a smooth transition that is applied to limit discontinuities in the data
+    that would appear in hard clipping. The transition phase is termed `knee`,
+    and two different types of knees can be applied (see parameters and
+    examples below).
+
+    Note that the limiting is applied on `signal.freq`, i.e. the data after
+    the FFT normalization in case `signal` is a `pyfar.Signal` object.
+
+    Parameters
+    ----------
+    signal : Signal, FrequencyData
+        The input data
+    limit : number, array like
+        The gain in dB at which the limiting reaches its full effect. If this
+        is a number, the same limit is applied to all frequencies. If this an
+        an array like, it must be broadcastable to `signal.freq.shape`.
+    knee : number, string
+        ``'arctan'``
+            Apply an arcus tangens knee according to [#]_ Section 3.6.4. In
+            this case the parameter `ratio` (below) is ignored. This knee
+            definition originates from microphone array signal processing.
+        number
+            Apply a knee with a width of `number` dB according to [#]_ Eq. (4).
+            This definition of the knee originates from the classic audio
+            effects limiting, compression, and expansion.
+
+    ratio : number, optional
+        Defines the ratio between the input and output level for frequencies
+        whose values exceed the `limit`. The default of ``np.inf`` strictly
+        limits all values. See examples below.
+    freq_range : array like, optional
+        Frequency range in which the limiting is applied. This must be an array
+        like containing the lower and upper limit in Hz. The default ``None``
+        applies the limiting to all frequencies.
+    direction : str, optional
+        Define how the limiting works
+
+        ``'above'`` (default)
+            Apply soft limiting to values above `limit`.
+        ``'below'``
+            Apply soft limiting to values below `limit`. This is achieved by
+            using ``1/signal.freq`` for the limiting.
```

Comment:
```suggestion
```
I'd remove this feature. This can be done in a second step outside of this function, giving users the option to use the inversion method of their choosing.

Reply:
Its not about inversion. This is only a trick to a signal to an absolute minimum value. I rephrased this. You can have a look at the example plots to see what it des.